src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -201,6 +203,29 @@ Chunk StorageObjectStorageSource::generate()
                 getUniqueStoragePathIdentifier(*configuration, reader.getObjectInfo(), false),
                 object_info.metadata->size_bytes, &filename);
 
+            if (!partition_columns.empty() && chunk_size && chunk.hasColumns())
```

Comment:
Let's move it to wrapper class for DataLake which will work on top of this source.